recipe/meta.yaml

```
@@ -21,29 +28,31 @@ requirements:
     - make  # [unix]
     - {{ compiler('c') }}
     - {{ compiler('cxx') }}
+    # pyarrow does not require nvcc but it needs to link against libraries in arrow-cpp=*=*cuda
+    - {{ compiler("cuda") }}  # [cuda_compiler_version not in (undefined, "None")]
   host:
     # directly pin boost-cpp as we also seem to directly include boost symbols
     # in the Python modules.
-    - arrow-cpp {{ version }}
+    - arrow-cpp ={{ version }}=*{{ build_ext }}
```

@h-vetinari , arrow-cpp-proc in `run_constraint` ought to handle the matching with `build_ext`.

---

I know, but the same information is in the build-string of `arrow-cpp`, making it wholly unnecessary to depend on the version (and build string) of `arrow-cpp-proc`. If you don't like it, then feel free to revert - I consider it cleaner/simpler.

---

This notation is probably the preferred one for `conda`.
```suggestion
    - arrow-cpp  {{ version }} *{{ build_ext }}
```

---

@pearu The `run_constraint` will only be applied if `arrow-cpp-proc` is installed. Currently it seems to me that when you don't have `arrow-cpp-proc` installed, you could install the `cpu` version of `arrow-cpp` while installing the CUDA version of `pyarrow`.

---

> This notation is probably the preferred one for `conda`.

Personally I dislike significant whitespace for this - I think it's very hard to read.

> Currently it seems to me that when you don't have arrow-cpp-proc installed, you could install the cpu version of arrow-cpp while installing the CUDA version of pyarrow.

That's IMO not possible because we're depending on the specific build version of `arrow-cpp` already.

---

> > This notation is probably the preferred one for `conda`.
> 
> Personally I dislike significant whitespace for this - I think it's very hard to read.

I didn't make the standard but I like to keep it :(

> > Currently it seems to me that when you don't have arrow-cpp-proc installed, you could install the cpu version of arrow-cpp while installing the CUDA version of pyarrow.
> 
> That's IMO not possible because we're depending on the specific build version of `arrow-cpp` already.

The comment was meant in regard to not having `arrow-cpp ={{ version }}=*{{ build_ext }}` in `run`. From my understanding, if that would be omitted, the mentioned situation could happen (or I'm missing another safeguard?)

---

> The comment was meant in regard to not having arrow-cpp ={{ version }}=*{{ build_ext }} in run. From my understanding, if that would be omitted, the mentioned situation could happen (or I'm missing another safeguard?)

No you're right, but not having *any* requirement on `arrow-cpp` in run broke the build, as both @pearu and I had to find out. So I don't think it can be omitted anyway, and then why shouldn't we also add the `build_ext`?

> I didn't make the standard but I like to keep it :(

It's not a hill I'm gonna die on, but that's IMO a bad reason to write hard-to-read recipes (and it's not like the variants are so different as to be unrecognisable).